components/sidebar/sidebar_channel/index.ts

```diff
@@ -26,16 +27,17 @@ function makeMapStateToProps() {
 
         const member = getMyChannelMemberships(state)[ownProps.channelId];
         const currentChannelId = getCurrentChannelId(state);
+        const collapsed = isCollapsedThreadsEnabled(state);
 
         // Unread counts
         let unreadMentions = 0;
         let unreadMsgs = 0;
         let showUnreadForMsgs = true;
         if (member) {
-            unreadMentions = member.mention_count;
+            unreadMentions = collapsed ? member.mention_count_root : member.mention_count;
 
             if (channel) {
-                unreadMsgs = Math.max(channel.total_msg_count - member.msg_count, 0);
+                unreadMsgs = collapsed ? Math.max(channel.total_msg_count_root - member.msg_count_root, 0) : Math.max(channel.total_msg_count - member.msg_count, 0);
```

Comment:
I think there were some requests on the other PR to make this logic into a selector. Would that be possible?